saleor/graphql/giftcard/resolvers.py

```diff
@@ -0,0 +1,19 @@
+import graphene
+
+from ...giftcard import models
+from ...graphql.core.utils import from_global_id_strict_type
+from ..core.utils import get_node_optimized
+from .types import GiftCard
+
+
+def resolve_gift_card(info, gift_card_global_id):
+    """Return gift card only for user assigned to it or proper staff user."""
+    user = info.context.user
+    if user.has_perm("giftcard.manage_gift_card"):
```

Comment:
From what I see the resolver is decorated in the schema with `@permission_required("giftcard.manage_gift_card")`, so what's the point of this check? Also, I don't understand why this resolver here uses `get_node_from_global_id` and later in the other case `get_node_optimized`.